### PR TITLE
Feature/wiki basics

### PR DESCRIPTION
**Pull request by open-schnick**

### CODE REVIEW

File: wiki/3_wiki.md [hunk 1]
```diff
@@ -95,9 +70,18 @@ Talking about security, all of your files will be only accessable with a passwor
...
+The admin will be responsible for creating all user accounts, the users them self will then be able to change some of their information like password and username. 
 
 
 
+For file interaction we have the basic use cases of viewing folder contents and be able to rename or delete files. The (technical) more advanced use cases are uploading, downloading and searching for files. One potential feature would be support for different media types like videos and music.
+
+To organize the permissions for files the users will be able to set view or editing permission for user or groups of users.
+
+ -->
+
```
Comment (open-schnick) [inline wiki/3_wiki.md:85]: I really like that part, i would keep it.

File: wiki/3_wiki.md [hunk 2]
```diff
@@ -133,43 +117,188 @@ The technologies we use are:
...
+To easily find files you will also be able to sort the files in the current folder by different values like filename or last edited.
+
+#### 3.1.7 Show own profile
+
+This will be site where the logged in user will be able to see his profile information, like his username, and the groups he is in.
+
+#### 3.1.8 Edit profile information (change password etc.)
+
+Once an account was created the user should change his password. He will also be able to change his username.
```
Comment (open-schnick) [inline wiki/3_wiki.md:176]: Why should the user change his password after registration?
How about:
`Once an account was created the user is able to change his username and password.`

File: wiki/3_wiki.md [hunk 3]
```diff
@@ -133,43 +117,188 @@ The technologies we use are:
...
+
+#### 3.1.9 Download files
+
+This is one of the most important use cases of the application. You will be able to download files you are allowed to see with your browser.
+
+
+
+#### 3.1.10 Upload files
+
+The same way you can download files you will also be able to upload files in places you are allowed to. This way you can also overwrite files.
```
Comment (open-schnick) [inline wiki/3_wiki.md:187]: But can the ui user really overwrite files? I think there should be at least a warning.

File: wiki/3_wiki.md [hunk 4]
```diff
@@ -133,43 +117,188 @@ The technologies we use are:
...
+
+#### 3.1.11 Search for files
+
+You will be able to search for files with a specific name within all the files you are allowed to view.
 
 
 ### 3.2 Usability
 
+We want our application to be easy to set up and easy to use. The user interface should be intuitive and self-explanatory. It should be close to different filesystem browsers you can find on operation systems.
+Setting up the application should be easy for everyone who has some experience with computers and servers. The installation will be done by a single shell script and require docker.
```
Comment (open-schnick) [inline wiki/3_wiki.md:197]: Better?:
`The installation will be done by executing one single shell script. And only requires docker as a dependency.`

File: wiki/3_wiki.md [hunk 5]
```diff
@@ -133,43 +117,188 @@ The technologies we use are:
...
 #### 3.3.1 Availability
 
-### 3.4 Perfomance
+Each user will be responsible for the availability of their own instance, but we should make sure there is always a stable version available for them to install.  
+
+#### 3.3.2 Data loss
+
+We shall provide options to back up all the data stored by our application. 
+
+
```
Comment (open-schnick) [inline wiki/3_wiki.md:216]: do we ? :D

File: wiki/3_wiki.md [hunk 6]
```diff
@@ -133,43 +117,188 @@ The technologies we use are:
...
+### 3.4 Performance
+
+Performance is important for all parts of our software and will always be considered during the development process.
+
```
Comment (open-schnick) [inline wiki/3_wiki.md:220]: Adding this?
`We also make sure, that our application can be run on very low level hardware like a raspberry pie. In the end our application will be self-hosted, so we cant make promises about performance on every hardware.`

File: wiki/3_wiki.md [hunk 7]
```diff
@@ -133,43 +117,188 @@ The technologies we use are:
...
 ### 3.5 Supportability
 
+#### 3.5.1 Coding Standards
+WIP
+We are going to write the code by using common clean code standards. For example, we will name our variables and methods by their functionalities. This will keep the code easy to read by everyone and make further development much easier.
+
+#### 3.5.2 Testing Strategy
+WIP
+The application will have a high test coverage and all important functionalities and edge cases should be tested. Further, mistakes in the implementation will be discovered instantly, and it will be easy to locate the error.
+
```
Comment (open-schnick) [inline wiki/3_wiki.md:244]: Better?
`The testable parts of our application e.g. RESTApi will have a high.....`

File: wiki/3_wiki.md [hunk 8]
```diff
@@ -133,43 +117,188 @@ The technologies we use are:
...
+Supported browsers will be:
+* recent versions of Chrome
+* recent versions of Firefox
+* recent versions of Safari
+
+
 ### 3.7 On-line User Documentation and Help System Requirements
+WIP
+The usage of the app should be as intuitive as possible, so it won't need any further documentation. If the user needs some help we will implement a "Help"-Button in the App which includes a FAQ.
+
```
Comment (open-schnick) [inline wiki/3_wiki.md:262]: do we?

Comment (qvalentin) [reply wiki/3_wiki.md:261]: we should

File: wiki/3_wiki.md (hunk 5 above)
Comment (qvalentin) [reply wiki/3_wiki.md:214]: Maybe just with a little shell script?
